src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1513,5 +1514,36 @@ private static void HandleBreakEvent(object? state)
                 }
             }
         }
+
+        public sealed class ReadOnlyMemoryContentComparer : IEqualityComparer<ReadOnlyMemory<char>>
+        {
+            public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y) =>
+                x.Span.SequenceEqual(y.Span);
+
+            public int GetHashCode(ReadOnlyMemory<char> obj)
+            {
+                // This hash code is a simplified version of some of the code in String,
```

Comment:
@stephentoub we use [Marvin.cs](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Marvin.cs) for `ReadOnlyMemory<byte>` hashcode computation. Wondering if a similar (non-randomized) implementation exists for char buffers and if not, would it make sense to move this to the common folder?

Reply:
See the comment I just added above. ;-)

Reply:
(That said, I would not be surprised if we find other places we want to use a ReadOnlyMemoryComparer like this, or even make it public.  If/when that happens, we can certainly move this.  Might make sense now to at least put it in its own file.)